include/networkit/centrality/PageRank.hpp

```diff
@@ -19,31 +23,53 @@ namespace NetworKit {
  * directed graphs; we follow the verbal description, which requires to sum over the incoming
  * edges (as opposed to outgoing ones).
  */
-class PageRank: public Centrality {
-protected:
-    double damp;
-    double tol;
+class PageRank final : public Centrality {
 
 public:
+    enum Norm { L1Norm, L2Norm };
+
     /**
      * Constructs the PageRank class for the Graph @a G
      *
      * @param[in] G Graph to be processed.
      * @param[in] damp Damping factor of the PageRank algorithm.
      * @param[in] tol Error tolerance for PageRank iteration.
      */
-    PageRank(const Graph& G, double damp=0.85, double tol = 1e-8);
+    PageRank(const Graph &G, double damp = 0.85, double tol = 1e-8);
 
     /**
      * Computes page rank on the graph passed in constructor.
      */
-    virtual void run();
+    void run() override;
 
     /**
-     * Returns upper bound on the page rank: 1.0. This could be tighter by assuming e.g. a star graph with n nodes.
+     * Returns upper bound on the page rank: 1.0. This could be tighter by assuming e.g. a star
+     * graph with n nodes.
      */
-    virtual double maximum();
+    double maximum() override;
+
+    /**
+     * Return the number of iterations performed by the algorithm.
+     *
+     * @return Number of iterations performed by the algorithm.
+     */
+    count numberOfIterations() const {
+        assureFinished();
+        return iterations;
+    }
+
+    // Maximum number of iterations allowed
+    count maxIterations = std::numeric_limits<count>::max();
+
+    // Norm used as stopping criterion
+    Norm norm = Norm::L2Norm;
```

Comment:
Not to familiar in with this topic, but is `L2Norm` the recommended norm for PageRank? And maybe it is a good idea to add example about different norms (I guess it only changes the time it takes to converge?) in the appropriate notebook.

Reply:
I think that both are valid choices, before we were using the `L2` norm so it makes sense to keep that as the default choice. But I agree that I should add an example in the centrality notebook.